expression/builtin_time.go

```diff
@@ -3416,6 +3416,11 @@ func (b *builtinAddDateStringStringSig) evalTime(row chunk.Row) (types.Time, boo
 		return types.ZeroTime, true, err
```

Comment:
VecEvalTime should also be modified, you can check the following example in MySQL and TiDB
``` sql
set @@sql_mode='';
create table t(a datetime);
insert into t values('0000-00-00 00:00:00');
select date_add(a, interval 2020 year) from t;
```

But it'll be ok to fix this in another PR.

Reply:
@XuHuaiyu @SunRunAway I think this vectorized function should also be fixed, but it may be too many changes for a single PR. To have a cleaner handling of this issue, a refactor of the code may be necessary. For example, I put the valid check in all the `add_date` functions, but I saw that there is _little differences_ among all the implemented functions. If you think it is better to fix it in the same PR, I will do it. But I will prefer to open a new one for it.